src/sources/transient.rs

```diff
@@ -267,11 +369,12 @@ impl<T: crate::EventSource> crate::EventSource for TransientSource<T> {
 mod tests {
```

Comment:
A bit of a nitpick, but the tests module is huge after this pull request, it's nearly 3x the size of the rest of the file.

Would it make sense to move the tests to their own file `src/sources/transient/tests.rs`?

Reply:
Fine by me. Arguably these are integration tests, not unit tests anyway. I'd happily put them in `/tests/`, but then it'd be a consistency issue with all the other not-actually-unit-tests.

Reply:
That's right that many of the tests in calloop can probably be moved to `/tests/`, but I suggest we leave that for another PR.